src/Cache/Mutator/AddHeaderResponseMutator.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace Http\Client\Common\Plugin\Cache\Mutator;
+
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+
+/**
+ * Adds a header if the response came from cache.
+ *
+ * @author Iain Connor <iain.connor@priceline.com>
+ */
+class AddHeaderResponseMutator implements ResponseMutator
+{
+    /** @var string */
+    private $headerName;
+
+    /**
+     * @param string $headerName
+     */
+    public function __construct($headerName = 'X-From-Php-Http-Cache')
```

Comment:
a semi-official way to mark cache hits is to use "X-Cache: HIT" (resp. MISS). can we find a different example?

Reply:
Good suggestion, updated.